### ISSUE TITLE
Not triggering UrlChanged on mount breaks pathMode routing

### ISSUE THREAD

**Issue by Dzoukr** [Zaid-Ajaj/Feliz.Router#16]
Hello friend,

I found a strange issue. When updating to `Feliz.Router` higher than `1.8.0`, the routing is broken for `Router.pathMode`. There are two issues:

1. Link created from `Router.format` is having hash instead of path (e.g. `http://localhost:8080/#/login` instead of `http://localhost:8080/login`)
2. When entering manually routed address (like `http://localhost:8080/login`), the routing does not work at all.

I suspect this commit to be the https://github.com/Zaid-Ajaj/Feliz.Router/commit/9da1d6723e06534072a67f0655a9469baa0c836d cause.

Is there any chance to have `UrlChanged` triggered on mount again, or do I need to stay with `1.8.0`?

**Comment by Zaid-Ajaj** [comment]
Hello Roman, looks like I broke something again 😅 sorry about that! 

Let me see, I don't remember changing `Router.format` so that is a bit weird, I will look into it. As for the initial trigger on start up, it was weird to have the application trigger a `UrlChanged` if the URL hasn't really changed so instead of that, you can ask for the current URL via `Router.currentUrl()` when you initialize the root program, have you done that by any chance?

**Comment by Dzoukr** [comment]
Hi Zaid,

no problem, it happens to me aaaaall the time. 😄 

I can try to plug `Router.currentUrl()`, but when I manually pass into browser some routed path (e.g. `http://localhost:8080/login`), shouldn't be `UrlChanged` triggered, because right after init I want to move to some existing sub-page so routing should happen? Or do I understand it wrong? Thanks for any help.

**Comment by Dzoukr** [comment]
Also `Router.currentUrl()` for `http://localhost:8080/some-sub-page`  gives me 

```javascript
List {head: undefined, tail: undefined}
head: undefined
tail: undefined
__proto__: Object
```

**Comment by Zaid-Ajaj** [comment]
Oh man 😞 this is bigger than I thought. The functions `Router.currentUrl`, `Router.format` and `Router.navigate` depend on the `Router.router` component being initialized to properly work because it is the one that sets the routing mode as a global variable. 

When you call these functions on initial render **before** the router component is initialized, they use the default global router mode which is the hash routing

> That is why `Router.currentUrl("http://localhost:8080/some-sub-page") == [ ]` in your case **initially** because it was looking for the segments after the hash sign but it isn't there

Then after just ONE message is dispatched, the router has already been initialized and the functions  `Router.currentUrl`, `Router.format` and `Router.navigate` will give correct results. This wasn't visible in version 1.8.0 or earlier because the router always caused message dispatch when it was initialized so it fixed itself before anything meaningful happens in the application but now the cracks have been opened. 

I used the global variable for the router mode because otherwise you would have to pass the mode to every call of `Router.currentUrl`, `Router.format` and `Router.navigate` or have variants like `Router.currentUrlPath()` and `Router.formatPath()` for path specific routes. This practically enables the user to set it only in one place and the rest of application works like it should but it sure has its downsides like we are seeing here 😭 

> "Just a tiny little bit of global state, only one flag, what could go wrong?" 

There are two solutions to this problem:
 1. Quick and simple fix -> call a function like `Router.enablePathMode()` at the entry point of the client before the Elmish program is bootstrapped which sets the global mode variable to use paths instead of hash. 
 2. Refactor the library to pass path mode to every call or have function variants that specialized for path routes. 

The first one is of course the easy and simple one but I don't really like it (side-effects 👎) but the second requires the user to do more work to get the job done. 

What do you think Roman? 

(cc @marcpiechura because you use path modes as well)

**Comment by Dzoukr** [comment]
Ufff... I think it's your call @Zaid-Ajaj, but nobody likes side-effects, right? 😄  If you would have time to make proper refactoring, it would be great. But no problem for me to keep using `1.8.0` for some time - from my perspective library works great!

**Comment by Zaid-Ajaj** [comment]
Fixed in v2.0 :rocket: please give it a shot! Here are [the new API docs](https://github.com/Zaid-Ajaj/Feliz.Router#using-path-routes-without-hash-sign)

**Comment by Dzoukr** [comment]
Works like a charm, thank you! 🚀 

Just for your information to have routing working even on hard refresh, you need to use `Router.getCurrentPath()` like this:


```fsharp
// Domain.fs
type Page =
    | Login
    | Calendar

type Model = {
    CurrentPage : Page
}

module Model =
    let init page = {
        CurrentPage = page
    }

// View.fs
let parseUrl = function // convert paths to Page
    | [ "login" ] -> Login
    | [ "calendar" ] -> Calendar
    | _ -> Calendar

// State.fs
let init () =
    // get currentPath righ after init 
    // parse url to Page
    // setup initial model with parsed Page as current one
    // PROFIT
    Model.init (Router.currentPath() |> View.parseUrl), Cmd.none
```
